Subject:    [PATCH 1/1] sg: Remove aha1542 hack
From: bjking1@us.ibm.com
Bug 148506 - LTC21395-iprconfig shows SCSI adapters as SAS adapters

Remove a hack in the sg driver that alters the total buffer
length for SG_IO commands to ensure buffers are not odd byte
lengths. This breaks on the ipr driver since it requires the
request_bufflen to equal the length specified in the cdb.
The block layer SG_IO code does not appear to have this hack.

Signed-off-by: Brian King <brking@us.ibm.com>
Signed-off-by: Olaf Hering <olh@suse.de>

 drivers/scsi/sg.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.16-rc2-olh/drivers/scsi/sg.c
===================================================================
--- linux-2.6.16-rc2-olh.orig/drivers/scsi/sg.c
+++ linux-2.6.16-rc2-olh/drivers/scsi/sg.c
@@ -2162,7 +2162,7 @@ sg_link_reserve(Sg_fd * sfp, Sg_request 
 
 	srp->res_used = 1;
 	SCSI_LOG_TIMEOUT(4, printk("sg_link_reserve: size=%d\n", size));
-	rem = size = (size + 1) & (~1);	/* round to even for aha1542 */
+	rem = size;
 
 	for (k = 0; k < rsv_schp->k_use_sg; ++k, ++sg) {
 		num = sg->length;
